### PR TITLE
Add "Do Not Merge" GH Action to block PRs with "DO-NOT-MERGE" label

### PR DESCRIPTION
**Pull request by digiwand**
## Explanation

backstory: 
I accidentally merged a PR with the "DO-NOT-MERGE" label today https://github.com/MetaMask/metamask-extension/pull/18652 😅

This PR blocks PRs with the "DO-NOT-MERGE" from being merged

Inspo for logic https://www.jessesquires.com/blog/2021/08/24/useful-label-based-github-actions-workflows/#updated-21-march-2022

## Screenshots/Screencaps

Screen Shot 2023-04-21 at 11 18 58 AM

## Manual Testing Steps

Apply or Remove "DO-NOT-MERGE" label and observe PR checks below

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: .github/workflows/do-not-merge.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+name: Check "DO-NOT-MERGE" label
+
+on:
+  pull_request:
+    types: [opened, reopened, labeled, unlabeled, synchronize]
+
+jobs:
+  do-not-merge:
+    runs-on: ubuntu-latest
+    if: ${{ contains(github.event.*.labels.*.name, 'DO-NOT-MERGE') }}
```
Comment (Gudahtt) [inline .github/workflows/do-not-merge.yml:12]: Nit: The only event trigger here is `pull_request`, so the GitHub event here should always be of this type: https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request

I think that means we can specify `pull_request` directly here in place of this first asterisk. We only need the second asterisk, to check every label entry in the array.

```suggestion
    if: ${{ contains(github.event.pull_request.labels.*.name, 'DO-NOT-MERGE') }}
```

**Review by Gudahtt — DISMISSED** [review]
LGTM, great idea! Just one minor suggestion

File: .github/workflows/do-not-merge.yml (hunk 1 above)
Comment (digiwand) [reply .github/workflows/do-not-merge.yml]: LGTM! check continues to work as expected

**Review by brad-decker — APPROVED** [review]
Awesome

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**